healthcheck-spring/src/main/java/com/storebrand/healthcheck/spring/HealthCheckRegistryFactory.java

```diff
@@ -82,19 +87,77 @@ public void stopHealthChecks() {
 
     public static class DefaultSlf4jHealthCheckLogger implements HealthCheckLogger {
         private static final Logger log = LoggerFactory.getLogger(DefaultSlf4jHealthCheckLogger.class);
+        private final ServiceInfo _serviceInfo;
+
+        public DefaultSlf4jHealthCheckLogger(ServiceInfo serviceInfo) {
+            _serviceInfo = serviceInfo;
+        }
 
         @Override
+        @SuppressWarnings("try")
+        // try: Using try-with-resources to ensure MDC context is cleaned up.
         public void logHealthCheckResult(HealthCheckDto dto) {
             // Setting max char length, so we don't spam logs with insane log lines. Health checks should not return
             // gigantic messages.
             int charLength = 12_000;
-            String body = HealthCheckTextOutput.getBodyFromHealthCheck(dto);
-            if (body.length() > charLength) {
-                body = body.substring(0, charLength) + " (...) CHOPPED by Health check logger. Above " + charLength
-                        + " chars limit.";
+            String activatedAxes = dto.axes.activated.stream()
+                    .map(Axis::name)
+                    .collect(Collectors.joining(","));
+
+            String runningTimeInMs = String.format("%.4f", dto.runStatus.runningTimeInNs / 1_000_000.0d);
+            // If onBehalfOf is set, we use it. If absent, we use the service name as the default onBehalfOf.
+            String onBehalfOf = dto.onBehalfOf.orElse(_serviceInfo.getServiceInfo().project.name);
+            try (MDCStack ignored = MDCStack.create("healthCheck.")
+                    .add("name", dto.name)
+                    .add("serviceName", _serviceInfo.getServiceInfo().project.name)
+                    .add("onBehalfOf", onBehalfOf)
+                    .add("version", _serviceInfo.getServiceInfo().project.version)
+                    .add("slow", Boolean.toString(dto.runStatus.slow))
+                    .add("crashed", Boolean.toString(dto.runStatus.crashed))
+                    .add("activatedAxes", activatedAxes)
+                    .add("runningTimeInMs", runningTimeInMs)) {
+                if (dto.runStatus.ok) {
+                    log.info("HealthCheck [" + dto.name + "] passed in [" + runningTimeInMs + "] ms.");
+                }
+                else {
+                    String body = HealthCheckTextOutput.getBodyFromHealthCheck(dto);
+                    if (body.length() > charLength) {
+                        body = body.substring(0, charLength) + " (...) CHOPPED by Health check logger. Above "
```

Comment:
we should have an MDC field for chopped, so we can check if a message was chopped. Only need to be added if we actually chop the message.